qemu/tests/nic_hotplug.py

```diff
@@ -122,7 +124,28 @@ def device_add_nic(pci_model, netdev, device_id):
         add_output = vm.monitor.send_args_cmd(pci_add_cmd)
         return add_output
 
+    def run_sub_test(params, plug_tag):
+        """
+        Run subtest before/after hotplug/unplug device.
+
+        :param plug_tag: identify when to run subtest,
+                         ex, before_hotplug.
+        :return: whether vm was successfully shut-down
+                 if needed
+        """
+        sub_type = params.get("sub_type_%s" % plug_tag)
+        if sub_type:
+            context_msg = "Running sub test '%s' %s"
+            error_context.context(context_msg % (sub_type, plug_tag),
+                                  logging.info)
+            utils_test.run_virt_sub_test(test, params, env, sub_type)
+            if sub_type == "shutdown" and vm.is_dead():
```

Comment:
@kimi1978wy Could " if sub_type == "shutdown" and vm.is_dead():" be replaced with "if sub_type == "shutdown" "? 
As when it shutdown fail, the case will fail at the previous line. if it success, the vm must is dead, so I guess maybe can not add this condition here. What do you think?